src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,128 @@
+//
+// ImageIO - CGImageAnimation.cs
+//
+// Authors:
+//	Whitney Schmidt  <whschm@microsoft.com>
+//
+// Copyright 2020, Microsoft Corp.
+//
+
+using System;
+using System.Runtime.InteropServices;
+using CoreFoundation;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace ImageIO {
+
+    public partial class CGImageAnimation {
+
+        public delegate void CGImageSourceAnimationBlock (nint arg0, CGImage arg1, out bool done /*check*/ );
+
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern unsafe int CGAnimateImageAtURLWithBlock (/* CFURLRef */ IntPtr url, /* CFDictionaryRef _iio_Nullable */ IntPtr options, /* CGImageSourceAnimationBlock */ IntPtr block);
```

Comment:
I believe we can pass the Block as a ref like we do here: https://github.com/xamarin/xamarin-macios/blob/master/src/Network/NWConnection.cs#L203

Would change the call you do later in:
```csharp
ret = CGAnimateImageAtURLWithBlock (url.Handle, options == null ? IntPtr.Zero : options.Handle, (IntPtr) block_ptr_block);
```
to
```csharp
ret = CGAnimateImageAtURLWithBlock (url.Handle, options == null ? IntPtr.Zero : options.Handle, ref block_block);
```

Once you have done that, we do not longer need the unsafe block:
```csharp
                BlockLiteral block_block;
                block_block = new BlockLiteral ();
                block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);

                try {
                    ret = CGAnimateImageAtURLWithBlock (url.Handle, options == null ? IntPtr.Zero : options.Handle, ref block_block);
                } finally {
                    block_block.CleanupBlock ();
                }
```

Reply:
updated! Thank you for the refresher on `unsafe`/`IntPtr`/passing by reference :)